.github/scripts/validate-deploy.sh

```diff
@@ -50,21 +50,25 @@ else
   sleep 30
 fi
 
-DEPLOYMENT="${COMPONENT_NAME}-${BRANCH}"
+RESOURCE01="db2oltp"
+
 count=0
-until kubectl get deployment "${DEPLOYMENT}" -n "${NAMESPACE}" || [[ $count -eq 20 ]]; do
-  echo "Waiting for deployment/${DEPLOYMENT} in ${NAMESPACE}"
+
+until kubectl get pods -l icpdsupport/addOnId=${RESOURCE01} -n "${NAMESPACE}" 1> /dev/null 2> /dev/null || [[ $count -eq 20 ]]; do
```

Comment:
Doing a `kubectl get pods -l` won't return an error code if it can't find any pods with the label so this check will always be true. You should check that the count is greater than zero instead.

Also, what is creating these pods? A StatefulSet or a Deployment?
Does the StatefulSet/Deployment not have the same label? If possible, it would be better to check for the existence of the StatefulSet/Deployment instead of the individual pods.

Reply:
ok. Got it. StatefulSet is being created. Have updated the command to check the existence of Statefulset